Bootstrapper/appleGuiceBootstrapper.cpp

```diff
@@ -198,73 +200,83 @@ void parseLine(string &headerEntry,
+                set<string> superProtocols = protocolToSuperProtocols[protocolName];
+                if (contains(superProtocols, NSObject)) {
+                    protocolToSuperProtocols[protocolName].erase(NSObject);
+                }
+                if (superProtocols.size() != 0) {
+                    nextRoundImplementedProtocols.insert(superProtocols.begin(), superProtocols.end());
+                }
+            }
+        }
+        implementedProtocols = nextRoundImplementedProtocols;
+    }
 }
 
 void protocolsImplementedByClass(string &className,
-				   unordered_map <string, string> &classToSuperClass,
-			   unordered_map <string, set<string> > &classToProtocols,
-			   set<string> &result) 
+                                 unordered_map <string, string> &classToSuperClass,
+                                 unordered_map <string, set<string> > &classToProtocols,
+                                 set<string> &result)
 {
-
-	if (contains(classToProtocols, className)) {
-		set<string> implementedProtocols = classToProtocols[className];
-		//must be a better way to union sets
-		for(set<string>::const_iterator it = implementedProtocols.begin(); it != implementedProtocols.end(); it++ ) {
-			string protocolName = *it;
-			result.insert(protocolName);
-		}
-	}
-
-	if (contains(classToSuperClass, className)) {
-		string superClass = classToSuperClass[className];
-		protocolsImplementedByClass(superClass, classToSuperClass, classToProtocols, result);
-	}
-}	
-
-void travereOverClassHierarchAndAct(string &className, unordered_map<string, set<string> > &protocolToSuperProtocols, 
-			   unordered_map <string, string> &classToSuperClass,
-			   unordered_map <string, set<string> > &classToProtocols,
-			   unordered_map <string, set<string> > &protocolToImps) 
-{
-	
-	set<string> implementedProtocols;
-	protocolsImplementedByClass(className, classToSuperClass, classToProtocols, implementedProtocols);
-	
-	addBindToResultList(className, implementedProtocols, protocolToSuperProtocols, protocolToImps);
-	
-	//cout << ">> " << className << " ";
-	
-	if (!contains(classToSuperClass, className)) {
-		//cout << "||" << endl;
-		return;
-	}
-
-	string superClass = classToSuperClass[className];
-	travereOverClassHierarchAndAct(superClass, protocolToSuperProtocols, classToSuperClass, classToProtocols, protocolToImps);
+    
+    bool stop = false;
+    string curClassName = className;
+    
+    while (!stop) {
+        if (contains(classToProtocols, curClassName)) {
+            set<string> implementedProtocols = classToProtocols[curClassName];
+            result.insert(implementedProtocols.begin(), implementedProtocols.end());
+        }
+        
+        if (contains(classToSuperClass, curClassName)) {
+            string superClass = classToSuperClass[curClassName];
+            if (!isNSObject(superClass) && superClass.find(UIPrefix) != 0 && superClass.find(NSPrefix) != 0) {
```

Comment:
You are blocking here any class that starts with NS or UI, assuming these are built in classes. What if a user has his own 3 letter prefix like `UIX` or `NSX'?

Reply:
Yeah I know. But it saves me some time.
Maybe we can maintain blacklist (configured by the user) of prefixes he wants to ignore?

Reply:
If you want such functionality, it should be opt in, not opt out. The default will process all prefixes, and you can list the prefixes you want to exclude from processing (UI, NS).

Reply:
Added an option to pass prefix parameters to ignore. 
Usage - 
$(SHELL) $(PODS_ROOT)/AppleGuice/Bootstrapper/appleGuiceBootstrapper.sh . ./tmpAgBindings *.framework* Pods UI,NS